src/frontend/src/optimizer/plan_node/logical_iceberg_scan.rs

```diff
@@ -74,9 +83,13 @@ impl Distill for LogicalIcebergScan {
 
 impl ColPrunable for LogicalIcebergScan {
     fn prune_col(&self, required_cols: &[usize], _ctx: &mut ColumnPruningContext) -> PlanRef {
-        // TODO: support column pruning for iceberg scan
-        let mapping = ColIndexMapping::with_remaining_columns(required_cols, self.schema().len());
-        LogicalProject::with_mapping(self.clone().into(), mapping).into()
+        if required_cols.is_empty() {
+            let mapping = ColIndexMapping::with_remaining_columns(required_cols, self.schema().len());
+            // If reuqiured_cols is empty, we use the first column of iceberg to avoid the empty schema.
+            LogicalProject::with_mapping(self.clone_with_required_cols(&[0]).into() , mapping).into()
```

Comment:
So iceberg scan does not support scan no fields? But the RW scan can do it (only the row count is meaningful in this case) 
```SQL
dev=> explain select 1 from t;
                QUERY PLAN
-------------------------------------------
 BatchExchange { order: [], dist: Single }
 └─BatchProject { exprs: [1:Int32] }
   └─BatchScan { table: t, columns: [] }
(3 rows)
```

Reply:
This is a very corner case, but I am wondering if Iceberg supports creating a table without any columns. It should be possible in PG
```SQL
postgres=# create table t();
CREATE TABLE
postgres=# \d+ t;
                                          Table "public.t"
 Column | Type | Collation | Nullable | Default | Storage | Compression | Stats target | Description
--------+------+-----------+----------+---------+---------+-------------+--------------+-------------
Access method: heap
```